framework.go

```diff
@@ -180,7 +181,21 @@ func (f *Framework) ApplyCluster(ctx context.Context, cluster *application.Clust
 		return nil, fmt.Errorf("failed to apply cluster resources: %v", err)
 	}
 
-	return f.WaitForClusterReady(ctx, cluster.Name, cluster.GetNamespace())
+	kubeClient, err := f.WaitForClusterReady(ctx, cluster.Name, cluster.GetNamespace())
+	if err != nil {
+		return nil, err
+	}
+
+	// Create the E2E test service account and create a new client authenticated as it
+	testClient, err := testuser.Create(ctx, kubeClient)
```

Comment:
Do we want to do the service account dance on for all providers?

Reply:
Yes, I think so, if for no other reason than consistency. 

The approach works for any and all providers so it makes sense to do it the same way as there's not really any down-side to it.

Reply:
Happy with that as long as it doesn't slow down cluster creation and setup even further.

Reply:
Nah, this takes less than a second.